etcdserver/v3_server.go

```diff
@@ -109,7 +131,14 @@ func (s *EtcdServer) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeRe
 }
 
 func (s *EtcdServer) Put(ctx context.Context, r *pb.PutRequest) (*pb.PutResponse, error) {
-	result, err := s.processInternalRaftRequest(ctx, pb.InternalRaftRequest{Put: r})
+	var result *applyResult
+	var err error
+	for {
```

Comment:
this loop is copied around in a lot of places; auth revision checking probably belongs in processInternalRaftRequest